client/memory.c

```diff
@@ -131,6 +134,10 @@ int xdag_mem_init(size_t size)
 void *xdag_malloc(size_t size)
 {
 	uint8_t *res;
+	
+	if (size <= 0) {
```

Comment:
size_t is unsigned anyway

Reply:
this piece of code was after return malloc(size) because it is unlikely that someone will ask to malloc 0. Now it will do this check even when you use don't use tmp file.. that is useless. But at least it isn't a bug :)